Sources/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -219,17 +225,24 @@ class DatabaseContainer: NSPersistentContainer {
     /// messages pedning sent. You can use this option to warn a user about potential data loss.
     ///   - completion: Called when the operation is completed. If the error is present, the operation failed.
     ///
-    func removeAllData(force: Bool = true) throws {
+    func removeAllData(force: Bool = true, completion: ((Error?) -> Void)? = nil) {
         if !force {
             fatalError("Non-force flush is not implemented yet.")
         }
         
-        sendNotificationForAllContexts(name: Self.WillRemoveAllDataNotification)
-
-        // If the current persistent store is a SQLite store, this method will reset and recreate it.
-        try recreatePersistentStore()
-
-        sendNotificationForAllContexts(name: Self.DidRemoveAllDataNotification)
+        writableContext.perform { [weak self] in
+            do {
+                self?.sendNotificationForAllContexts(name: Self.WillRemoveAllDataNotification)
+                
+                // If the current persistent store is a SQLite store, this method will reset and recreate it.
+                try self?.recreatePersistentStore { error in
+                    self?.sendNotificationForAllContexts(name: Self.DidRemoveAllDataNotification)
+                    completion?(error)
+                }
+            } catch {
+                completion?(error)
+            }
+        }
```

Comment:
In this particular case, I wouldn't weakify self and let the DB flushing process complete.